lib/mongo/client.rb

```diff
@@ -95,7 +95,10 @@ class Client
     def_delegators :@database, :command, :collections
 
     # Delegate subscription to monitoring.
-    def_delegators :@monitoring, :subscribe, :unsubscribe
+    def_delegators :monitoring, :subscribe, :unsubscribe
+
+    # Delegate monitoring to cluster.
+    def_delegators :cluster, :monitoring
```

Comment:
I was running into the same command monitoring issue I ran into in the transactions tests (namely that creating a new client with `with` left the client in the state where its `monitoring` instance was different than `cluster`'s. I realized that since they are always supposed to be the same, the cleanest solution is to have clients just not have their own instance and delegate to the clusters.

Reply:
:+1: